### PR TITLE
Break expression dependency on const folder

### PR DESCRIPTION
**Pull request by RazvanN7**
I don't see the point in having the function defined in another module just to have another indirection.

### CODE REVIEW

**Review by dkorpel — APPROVED** [review]
Nice one!

File: compiler/src/dmd/expression.d [hunk 1]
```diff
@@ -1358,7 +1356,21 @@ extern (C++) abstract class Expression : ASTNode
...
+        switch (op)
+        {
+        case EXP.int64:
+        case EXP.float64:
+        case EXP.complex80:
+            return 1;
+        case EXP.null_:
+            return 0;
+        case EXP.symbolOffset:
+            return 2;
```
Comment (SixthDot) [inline compiler/src/dmd/expression.d:1369]: is `2` vs `1` still used nowadays ?

File: compiler/src/dmd/expression.d [hunk 2]
```diff
@@ -1358,7 +1356,21 @@ extern (C++) abstract class Expression : ASTNode
 
     final int isConst()
```
[the remark this answers is not in the document]
Comment (RazvanN7) [reply compiler/src/dmd/expression.d:1357]: Agree but that can be done in a future PR

File: compiler/src/dmd/expression.d (hunk 1 above)
Comment (RazvanN7) [reply compiler/src/dmd/expression.d:1369]: Nowadays, probably not, but when this code was written....

Comment (SixthDot) [reply compiler/src/dmd/expression.d:1369]: At least for clarity the return should be an enum member. I understand that this is out of the scope this PR tho.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**